Sources/OAuth2Client/NXOAuth2Request.m

```diff
@@ -105,6 +122,15 @@ - (void)performRequestWithSendingProgressHandler:(NXOAuth2ConnectionSendingProgr
     
     NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:self.resource];
     [request setHTTPMethod:self.requestMethod];
+    
+    if (self.customHeaderFields) {
+        [self.customHeaderFields enumerateKeysAndObjectsUsingBlock:^(NSString *field, NSString *value, BOOL *stop) {
+            if ([field isKindOfClass:NSString.class] && [value isKindOfClass:NSString.class]) {
```

Comment:
I would rather `NSAssert()` the class type here instead of silently failing if it's wrong. Since I would consider not passing `NSString` here for both a programmer error, which warrants an exception in objc.